lib/LTP/WhiteList.pm

```diff
@@ -129,6 +140,10 @@ sub find_whitelist_testsuite {
     return undef unless defined($path) and -e $path;
 
     my $content = path($path)->slurp;
+    my (undef, undef, $suffix) = fileparse($path, '\.[^\.]*');
+    if ($suffix ne '.json') {
+        $content = yaml2json($content);
+    }
     my $issues = Mojo::JSON::decode_json($content);
```

Comment:
This PR is needlessly compilcated. Just save the raw data to `ulogs` in `download_whitelist()` and add an `if/else` above:

    if ($suffix eq '.json') {
      $issues = decode_json($content);
    } elsif ($suffix eq '.yaml') {
      $issues = $yp->load_string($content);
    } else {
      die 'Invalid format';
    }

The data structure is the same in both YAML and JSON.

Reply:
Just utf8 decoding missing.
`$issues = $yp->load_string(decode_utf8 $content);`
or
`$yp->load_file($path)` will do the decoding automatically

Reply:
No, again :). Thx!